features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/SSLCertMonitor.java

```diff
@@ -222,27 +226,52 @@ public PollStatus poll(final MonitoredService svc, final Map<String, Object> par
                 for (int i = 0; i < certs.length && !serviceStatus.isAvailable(); i++) {
                     if (certs[i] instanceof X509Certificate) {
                         X509Certificate certx = (X509Certificate) certs[i];
+                        String subject = "";
+                        if (certx.getSubjectDN() != null && certx.getSubjectDN().getName() != null) {
+                            certx.getSubjectDN().getName();
+                        }
+                        String issuer = "";
+                        if (certx.getIssuerDN() != null && certx.getIssuerDN().getName() != null) {
+                            certx.getIssuerDN().getName();
+                        }
+                        String fprint = DatatypeConverter.printHexBinary(MessageDigest.getInstance("SHA-1").digest(certx.getEncoded())).toLowerCase();
+                        StringBuilder reasonBuilder = new StringBuilder();
                         LOG.debug("Checking validity against dates: [current: {}, valid: {}], NotBefore: {}, NotAfter: {}", calCurrent.getTime(), calValid.getTime(), certx.getNotBefore(), certx.getNotAfter());
                         calBefore.setTime(certx.getNotBefore());
                         calAfter.setTime(certx.getNotAfter());
                         if (calCurrent.before(calBefore)) {
-                            LOG.debug("Certificate is invalid, current time is before start time");
-                            serviceStatus = PollStatus.unavailable("Certificate is invalid, current time is before start time");
+                            reasonBuilder.append("Certificate with fingerprint '").append(fprint).append("' issued to ")
+                                    .append(subject).append(" by ").append(issuer)
+                                    .append(" is not yet valid. Current time is before start time. It is valid from ")
+                                    .append(certx.getNotBefore().toString()).append(" until ").append(certx.getNotAfter()).append(".");
```

Comment:
Guess we should also check these elements to be non-null, `certx.getNotBefore().toString()` ,  `certx.getNotAfter()`

Reply:
OK, I'm now just marking the service unavailable if either of them are null prior to attempting to use them. I assume that's sensible?

Reply:
yeah, that seems fine to me.